neuro_san/internals/run_context/langchain/toolbox_factory.py

```diff
@@ -56,87 +60,97 @@ class BaseToolFactory(ContextTypeBaseToolFactory):
-            self.base_tool_infos = self.overlayer.overlay(self.base_tool_infos, extra_base_tool_infos)
-
-            self.base_tool_info_file = base_tool_info_file
-
-    def create_base_tool(self, tool_name: str, user_args: Dict[str, Any] = None) -> Union[BaseTool, List[BaseTool]]:
+        restorer = ToolboxInfoRestorer()
+        self.toolbox_infos = restorer.restore()
+
+        # Mix in user-specified toolbox info, if available.
+        toolbox_info_file: str = os.getenv("AGENT_TOOLBOX_INFO_FILE")
+        if toolbox_info_file is not None and len(toolbox_info_file) > 0:
+            extra_toolbox_infos: Dict[str, Any] = restorer.restore(file_reference=toolbox_info_file)
+            self.toolbox_infos = self.overlayer.overlay(self.toolbox_infos, extra_toolbox_infos)
+
+            self.toolbox_info_file = toolbox_info_file
+
+    def create_tool_from_toolbox(
+            self,
+            tool_name: str,
+            user_args: Dict[str, Any] = None
+    ) -> Union[BaseTool, Dict[str, Any], List[BaseTool]]:
         """
         Resolves dependencies and instantiates the requested tool.
 
         :param tool_name: The name of the tool to instantiate.
         :param user_args: Arguments provided by the user, which override the config file.
         :return: - Instantiated tool if "class" of tool_name points to a BaseTool class
                  - A list of tools if "class of "tool_name points to a BaseToolkit class.
+                 - A dict of tool's "description" and "parameters" if tool_name points to a CodedTool
         """
         empty: Dict[str, Any] = {}
 
-        tool_info: Dict[str, Any] = self.base_tool_infos.get(tool_name)
+        tool_info: Dict[str, Any] = self.toolbox_infos.get(tool_name)
         if not tool_info:
-            raise ValueError(f"Tool '{tool_name}' is not defined in {self.base_tool_info_file}.")
+            raise ValueError(f"Tool '{tool_name}' is not defined in {self.toolbox_info_file}.")
 
         if not isinstance(tool_info, Dict):
             raise ValueError(f"The value for the {tool_name} key must be a dictionary.")
 
         if "class" not in tool_info:
             raise ValueError(
                 "Missing required key: 'class'.\n"
-                "Each tool configuration must include a 'class' key specifying the fully qualified class name.\n"
-                "Example: 'class': 'langchain_community.tools.bing_search.BingSearchResults'."
+                "Each tool must include a 'class' key:\n"
+                "- For Langchain base tools: use the full class path "
+                "(e.g., 'langchain_community.tools.bing_search.BingSearchResults')\n"
+                "- For shared CodedTools: use 'module.Class' format (e.g., 'websearch.WebSearch')"
             )
 
+        # If "description" in the tool info, then it is a shared coded tool.
+        # Return dictionary of tool's description and parameters.
+        if "description" in tool_info:
+            return tool_info
```

Comment:
In the future we might want to be able to override the description and/or other aspects of the @tool annotation so as to better integrate the tool into the agent network that is using it.  Certainly don't need to do that now.